README.md

```diff
@@ -1,44 +1,50 @@
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+### Testing :white_check_mark:
 
-### `npm run build`
+* The testing suite has been configured and setup. You just need to write your tests.
+* A combination of [JEST](https://jestjs.io/) and [React Testing library](https://testing-library.com/docs/react-testing-library/intro) should be used for component testing.
+* [Cypress](https://www.cypress.io/) will be used for end-to-end testing.
+* Pull Request for individual components will not be merged with passing JEST tests.
+* Pull Request for completed features will not be merged without end-to-end testing with Cypress.
 
-Builds the app for production to the `build` folder.<br>
-It correctly bundles React in production mode and optimizes the build for the best performance.
+### Project structure :card_file_box:
 
-The build is minified and the filenames include the hashes.<br>
-Your app is ready to be deployed!
+* The project directories has been setup using official react guidelines found on [this page](https://reactjs.org/docs/faq-structure.html). Please be sure to follow to directory structure to ensure that everything can be located by at a single glance.
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+## Tooling :wrench:
 
-### `npm run eject`
+The follwing tool chain has been chosen keeping in mind the speed of development, code quality, validations, security and a minimalistic bundle size. Please DO NOT add anything else.
 
-**Note: this is a one-way operation. Once you `eject`, you can’t go back!**
+I have not yet decided on wether to use **Redux** or not. I'm more inclined towards **MobX**. If you can come up with a valid arguement as to why we should use **Redux**, I'm all ears.
 
-If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
-
-Instead, it will copy all the configuration files and the transitive dependencies (Webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
-
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
-
-## Learn More
-
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
-
-To learn React, check out the [React documentation](https://reactjs.org/).
+* TypeScript
+* React Storybook
+* React Router
+* Formik
+* Yup
+* Styled Components
```

Comment:
Again, either pick styled components, or tailwind. Pick only one CSS solution. There is no point in having multiple ways to manage CSS in a single project.

Reply:
Will pick styled components. I was also thinking about emotion instead of Styled Components. I've heard in a few talks that it's lighter than Styled components.

Reply:
Cool.